sonar-core/src/main/java/org/sonar/core/util/Protobuf.java

```diff
@@ -169,4 +171,41 @@ protected void doClose() {
       IOUtils.closeQuietly(input);
     }
   }
+
+  /**
+   * Call a setter method of {@link com.google.protobuf.GeneratedMessage.Builder} if the parameter
+   * is not null. Do nothing if parameter is null.
+   * <br/>
+   * This utility method is convenient as the setter methods of Protobuf 2 do not accept
+   * {@code null} parameters. It avoid increasing complexity with "if not null" conditions.
+   * <br/>
+   * Example:
+   * <pre>
+   * setNullable(dto.getLine(), issueBuilder::setLine);
+   * </pre>
+   */
+  public static <PARAM> void setNullable(@Nullable PARAM parameter, Function<PARAM, ?> setter) {
+    if (parameter != null) {
+      setter.apply(parameter);
+    }
+  }
+
+  /**
+   * Same as {@link #setNullable(Object, Function)} but the parameter is converted by the function "{@code paramConverter}"
+   * before being applied to setter. If the converter returns {@code null}, then setter method
+   * is not called.
+   * <br/>
+   * Example:
+   * <pre>
+   * setNullable(dto.getIssueCreationDate(), issueBuilder::setCreationDate, DateUtils::formatDateTime);
+   * </pre>
+   * @see #setNullable(Object, Function)
+   */
+  public static <PARAM, PARAM2> void setNullable(@Nullable PARAM param, Function<PARAM2, ?> setter,
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Rename this generic name to match the regular expression '^[A-Z][0-9]?$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS00119)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Rename this generic name to match the regular expression '^[A-Z][0-9]?$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS00119)